src/sys/stat.rs

```diff
@@ -402,12 +402,69 @@ pub fn lutimes<P: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
+/// Timestamp update operation specifier used in [`utimensat`] and [`futimens`].
+#[cfg(not(target_os = "redox"))] // Redox does not support `UTIME_OMIT` and `UTIME_NOW`
+#[derive(Copy, Clone, Debug)]
+pub enum TimestampSpec {
+    /// Leave the timestamp unchanged.
+    Omit,
+    /// Update the timestamp to `Now`
+    Now,
+    /// Update the timestamp to the specific value specified in `TimeSpec`.
+    Set(TimeSpec),
+}
+
+/// Construct a new [`libc::timespec`] instance according to the value of [`TimestampSpec`].
+///
+/// This is used in the implementations of [`futimens`] and [`utimensat`]. As an
+/// end user, you may NOT find this trait implementation useful.
+///
+/// [`timespec`]: libc::timespec
+#[cfg(not(target_os = "redox"))] // Redox does not support `UTIME_OMIT` and `UTIME_NOW`
+impl From<&TimestampSpec> for libc::timespec {
+    fn from(spec: &TimestampSpec) -> Self {
+        let mut default_value = super::time::zero_init_timespec();
```

Comment:
With the use of this `zero_init_timespec()` function, I guess we don't need to take care of that padding field issue on `musl`?

Reply:
Yep